### PR TITLE
Fixing upload-artifact codeql debug step

### PR DESCRIPTION
**Pull request by joshjohanning**
### Why:

The current debug documentation under the 'Inspect the copy of the source files in the CodeQL database' heading doesn't work:

> Error: Invalid pattern '../codeql-database'. Relative pathing '.' and '..' is not allowed.

Also, it's `codeql-databases` not `codeql-database`.

Also fixing a missing `yaml` code block tag. 

### What's being changed:

Changing this:

```yaml
- uses: actions/upload-artifact@v2
  with:
    name: codeql-database
    path: ../codeql-database
```

To this:

```yaml
- uses: actions/upload-artifact@v2
  with:
    name: codeql-database-${{ matrix.language }}
    path: ${{ runner.temp }}/codeql_databases
```

### Check off the following:

- [ ] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@joshjohanning 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Comment by mchammer01** [comment]
Hey @joshjohanning 👋🏻 - thanks for creating this PR and improving our docs ✨ 
Your changes LGTM but I am no CodeQL expert so will ask someone from the CodeQL team to review.

File: content/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow.md [hunk 1]
```diff
@@ -140,11 +140,11 @@ For more information, see "[Configuring the {% data variables.product.prodname_c
...
 ### Inspect the copy of the source files in the {% data variables.product.prodname_codeql %} database
 You may be able to understand why some source files haven't been analyzed by inspecting the copy of the source code included with the {% data variables.product.prodname_codeql %} database. To obtain the database from your Actions workflow, add an `upload-artifact` action after the analysis step in your code scanning workflow:
-```
+```yaml
 - uses: actions/upload-artifact@v2
   with:
-    name: codeql-database
-    path: ../codeql-database
+    name: {% raw %}codeql-database-${{ matrix.language }}{% endraw %}
+    path: {% raw %}${{ runner.temp }}/codeql_databases{% endraw %}
```
Comment (erik-krogh) [inline content/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow.md:147]: I don't get these `{% raw %}` and `{% endraw %}`.  
They don't seem to be interpreted by the GitHub markdown renderer, so they appear in the rendered markdown (and I don't think that was your intent). 

The `${{ matrix.language }}` is rendered correctly regardless. 

```suggestion
    name: codeql-database-${{ matrix.language }}
    path: ${{ runner.temp }}/codeql_databases
```

**Comment by mchammer01** [comment]
Thanks for the review @erik-krogh 🙇🏻 - I've accepted the suggestion and will take a look at Staging to see how it renders.

**Comment by erik-krogh** [comment]
> Thanks for the review @erik-krogh 🙇🏻 - I've accepted the suggestion and will take a look at Staging to see how it renders.

Oh, I didn't notice the staging. Maybe the raw tags are needed after all, I just checked the GitHub markdown rendering [here](https://github.com/github/docs/blob/784d27e51b06c923f952511e0e45ed54d7ed8531/content/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow.md).  

The code looked fine otherwise.

File: content/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow.md [hunk 2]
```diff
@@ -140,11 +140,11 @@ For more information, see "[Configuring the {% data variables.product.prodname_c
 
 ### Inspect the copy of the source files in the {% data variables.product.prodname_codeql %} database
 You may be able to understand why some source files haven't been analyzed by inspecting the copy of the source code included with the {% data variables.product.prodname_codeql %} database. To obtain the database from your Actions workflow, add an `upload-artifact` action after the analysis step in your code scanning workflow:
-```
+```yaml
 - uses: actions/upload-artifact@v2
   with:
-    name: codeql-database
-    path: ../codeql-database
+    name: codeql-database-${{ matrix.language }}
```
Comment (mchammer01) [inline content/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow.md:146]:
```suggestion
    name: {% raw %}codeql-database-${{ matrix.language }}{% endraw %}
```

File: content/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow.md [hunk 3]
```diff
@@ -140,11 +140,11 @@ For more information, see "[Configuring the {% data variables.product.prodname_c
...
+    path: ${{ runner.temp }}/codeql_databases
```
Comment (mchammer01) [inline content/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow.md:147]:
```suggestion
    path: {% raw %}${{ runner.temp }}/codeql_databases{% endraw %}
```

**Comment by mchammer01** [comment]
Yes, I think those tags are needed. I'll re-add them and double-check on Staging.

**Review by mchammer01 — APPROVED** [review]
LGTM ✨ - and thanking you very much for your patience @joshjohanning 💖

**Comment by adityasharad** [comment]
Apologies for not seeing this earlier. The instructions can be made even simpler: https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/troubleshooting-the-codeql-workflow#creating-codeql-debugging-artifacts is sufficient for the Action to upload a database for debugging, and there is no longer any need for a separate call to `actions/upload-artifact@v2` in order to upload the database.

**Pull request merged**